### ISSUE TITLE
Error with profile listener: Missing or insufficient permissions. Error: Missing or insufficient permissions.

### ISSUE THREAD

**Issue by wallacepeng** [prescottprue/react-redux-firebase#508]
When i logout from firebase, i got below errors

Error with profile listener: Missing or insufficient permissions. Error: Missing or insufficient permissions.
    at new FirestoreError (index.cjs.js:346)
    at JsonProtoSerializer../node_modules/@firebase/firestore/dist/index.cjs.js.JsonProtoSerializer.fromRpcStatus (index.cjs.js:5258)
    at JsonProtoSerializer../node_modules/@firebase/firestore/dist/index.cjs.js.JsonProtoSerializer.fromWatchChange (index.cjs.js:5756)
    at PersistentListenStream../node_modules/@firebase/firestore/dist/index.cjs.js.PersistentListenStream.onMessage (index.cjs.js:13652)
    at index.cjs.js:13597
    at PersistentListenStream.<anonymous> (index.cjs.js:13573)
    at step (tslib.es6.js:91)
    at Object.next (tslib.es6.js:72)
    at tslib.es6.js:65
    at new Promise (<anonymous>)
    at Object.__awaiter (tslib.es6.js:61)
    at index.cjs.js:13569
    at index.cjs.js:13045

**Comment by lhas2** [comment]
Hi @wallacepeng ! I'm receiving the same error. Did you find a solution for this?

**Comment by wallacepeng** [comment]
@lhas2 This is caused by using user profile for login with redirection handling enabled. You can workaround it by setting UserProfile to null in the config. 
like 

const rrfConfig = {
  userProfile: null,
...
...
};

**Comment by prescottprue** [comment]
Duplicate of #494. Error logs are just warnings (though they use console.error) and can be ignored as they don't impact functionality. Going to look into a fix this weekend.

**Comment by wallacepeng** [comment]
thanks.

**Comment by MoSwilam** [comment]
I am still getting the same exact error, and it can't be ignored as any reload prevents the app from reading information from firestore, any workaround yet ? thanks !

**Comment by prescottprue** [comment]
@MoSwilam What version are you using? Have you tried the `logErrors: false` config option?

**Comment by prescottprue** [comment]
@debaosuidecl `logErrors: false` should be passed as part of your config into the store enhancer ( `reactReduxFirebase`) when creating your store:

```js
const createStoreWithFirebase = compose(
  reactReduxFirebase(firebase, rrfConfig), // firebase instance as first argument
  // reduxFirestore(firebase) // <- needed if using firestore
)(createStore)
```

@MoSwilam have you tried this? We may want to open a new issue with repro steps and versions

**Comment by dejanstrancar** [comment]
I made it working without that error, with updating Firebase database rules section to:
```
service cloud.firestore {
  match /databases/{database}/documents {
    match /{document=**} {
      allow read, write: if request.auth.uid != null;
    }
  }
}
```

**Comment by 411c0d3d** [comment]
calling `Firebase.logout() ` after `firebase.auth().signOut() `  fixed the problem for me :

**Comment by arroyoruy** [comment]
> calling `Firebase.logout() ` after `firebase.auth().signOut() ` fixed the problem for me :

me too, thanks!

**Comment by esbenvb** [comment]
@411c0d3d 

> calling `Firebase.logout() ` after `firebase.auth().signOut() ` fixed the problem for me :

Thanks it also did the trick here. However, I wonder why this works. The type defs of my `firebase` module (5.8.6) does not include the `logout()` method, and neither does the [official documentation](https://firebase.google.com/docs/reference/js/firebase).

I still believe that it is a bug, that you have to call an undocumented function in order to avoid errors.

What also makes me wonder, is that I can't catch the permission error returned when I sign out.

Here's how my code looks. I belive that `handleFacebookLogout` should catch the permission error but it doesn't.

```tsx
  handleFacebookLogout = async () => {
    try {
      await firebase.auth().signOut()
      firebase.logout()
    } catch (e) {
      console.log('SIGNOUT ERROR', e)
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <Text>UserProfileScreen</Text>
        <LoginButton onLogoutFinished={this.handleFacebookLogout} />
      </View>
    )
  }
```

**Comment by prescottprue** [comment]
There should for sure be improved documentation as well as improved typings that include this.

Also, the logout method itself should handle calling `firebase.auth().signOut()` internally, so I'm thinking there is still something wrong here, so going to reopen.

**Comment by olufemi424** [comment]
`firebase.logout()` thats all you need...

**Comment by beautifwhale** [comment]
This is still an issue.

**Comment by prescottprue** [comment]
Fix [released in v3.1.0](https://github.com/prescottprue/react-redux-firebase/releases/tag/v3.1.0) - thanks to @illuminist for the PR!

**Comment by sincness** [comment]
The tip or takeout for me was that you not should make a redirect on your signOut function.
Seems to work just fine without it! :rocket:
